bootstrap/configure.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+PLUGIN_DIR=$1
+PLUGIN_NAME=$2
+PLUGIN_PATH=$3
+
+# Try to clean-up previous runs
+vault auth disable "${PLUGIN_PATH}"
+vault plugin deregister "${PLUGIN_NAME}"
+killall "${PLUGIN_NAME}"
+
+# Give a bit of time for the binary file to be released so we can copy over it
+sleep 3
+
+# Copy the binary so text file is not busy when rebuilding & the plugin is registered
+cp ./bin/"$PLUGIN_NAME" "$PLUGIN_DIR"
+
+# Sets up the binary with local changes
+vault plugin register \
+    -sha256="$(shasum -a 256 "$PLUGIN_DIR"/"$PLUGIN_NAME" | awk '{print $1}')" \
+    auth "${PLUGIN_NAME}"
+
+vault auth enable -path="${PLUGIN_PATH}" "${PLUGIN_NAME}"
+
+# Write the azure auth configuration
+vault write auth/"${PLUGIN_PATH}"/config \
+    tenant_id="${TENANT_ID}" \
+    client_id="${CLIENT_ID}" \
+    client_secret="${CLIENT_SECRET}" \
+    resource="https://management.azure.com/"
+
+# Write a role with some bound constraints
+vault write auth/"${PLUGIN_PATH}"/role/dev-role \
+    bound_subscription_ids="${SUBSCRIPTION_ID}" \
+    bound_resource_groups="${RESOURCE_GROUP_NAME}"
+    
+# Login using the access token
+vault write auth/"${PLUGIN_PATH}"/login \
+    role="dev-role" \
+    jwt="${ACCESS_TOKEN_JWT}" \
+    subscription_id="${SUBSCRIPTION_ID}" \
+    resource_group_name="${RESOURCE_GROUP_NAME}" \
+    vm_name="${VM_NAME}"
```

Comment:
I could probably comment this out but leave it as an example. I _think_ it'll be helpful to see a full working example here. Let me know!

Reply:
I don't think it hurts to leave this here. It is nice to have the full example